### PR TITLE
 add escapeshellarg to  nagios_bin binary passed to shell_exec

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

From @mhaskar 's PR https://github.com/centreon/centreon/pull/7663 -> rebased and conflict resolved.
PR already reviewed !

To be cherry-picked on the 18.10.x and 19.04.x branches after QA validation. For 2.8.x, an other PR is needed.

https://shells.systems/centreon-v19-04-remote-code-execution-cve-2019-13024

**Fixes** # (CVE)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x -> other PR
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/configuration/configGenerate/xml/generateFiles.php [hunk 1]
```diff
@@ -190,8 +190,10 @@ function printDebug($xml, $tabs)
 {
     global $pearDB, $ret, $centreon, $nagiosCFGPath;
 
-    $DBRESULT_Servers = $pearDB->query("SELECT `nagios_bin` FROM `nagios_server` " .
-        "WHERE `localhost` = '1' ORDER BY ns_activate DESC LIMIT 1");
+    $DBRESULT_Servers = $pearDB->query(
+        "SELECT `nagios_bin` FROM `nagios_server`
```
Comment (vhr) [inline www/include/configuration/configGenerate/xml/generateFiles.php]:
```suggestion
        "SELECT `nagios_bin` FROM `nagios_server`"
```

File: www/include/configuration/configGenerate/xml/generateFiles.php [hunk 2]
```diff
@@ -190,8 +190,10 @@ function printDebug($xml, $tabs)
...
+        WHERE `localhost` = '1' ORDER BY ns_activate DESC LIMIT 1"
```
Comment (vhr) [inline www/include/configuration/configGenerate/xml/generateFiles.php]:
```suggestion
        . "WHERE `localhost` = '1' ORDER BY ns_activate DESC LIMIT 1"
```

File: www/include/configuration/configGenerate/xml/generateFiles.php [hunk 3]
```diff
@@ -209,7 +211,8 @@ function printDebug($xml, $tabs)
 
     foreach ($tab_server as $host) {
         $stdout = shell_exec(
-            $nagios_bin["nagios_bin"] . " -v " . $nagiosCFGPath . $host["id"] . "/centengine.DEBUG 2>&1"
+            escapeshellarg($nagios_bin["nagios_bin"]) . " -v " . $nagiosCFGPath .
```
Comment (vhr) [inline www/include/configuration/configGenerate/xml/generateFiles.php]:
```suggestion
            escapeshellarg($nagios_bin['nagios_bin']) . ' -v ' . $nagiosCFGPath .
```

File: www/include/configuration/configGenerate/xml/generateFiles.php [hunk 4]
```diff
@@ -209,7 +211,8 @@ function printDebug($xml, $tabs)
...
+            escapeshellarg($host["id"]) . "/centengine.DEBUG 2>&1"
```
Comment (vhr) [inline www/include/configuration/configGenerate/xml/generateFiles.php]:
```suggestion
            escapeshellarg($host['id']) . '/centengine.DEBUG 2>&1'
```

**Comment by sc979** [comment]
Succesfully cherry-picker on the branches :
- 19.04.x
- 18.10.x
- 2.8.x -> with conflicts resolution or adaptation

**Pull request merged**